src/sentry/rules/history/preview.py

```diff
@@ -366,27 +398,36 @@ def apply_frequency_conditions(
 
 
 def get_frequency_buckets(
-    project: Project, start: datetime, end: datetime, group_id: str
+    project: Project,
+    start: datetime,
+    end: datetime,
+    group_id: int,
+    dataset: Dataset,
 ) -> Sequence[Dict[str, Any]]:
     """
     Puts the events of a group into buckets, and returns the bucket counts.
     """
-    # TODO: support counting of other issue types and fields (# of unique users, ...)
-    bucket_counts: Sequence[Dict[str, Any]] = raw_query(
-        dataset=Dataset.Events,
-        start=start,
-        end=end,
-        filter_keys={"project_id": [project.id]},
-        conditions=[("group_id", "=", group_id)],
-        aggregations=[
+    if dataset is None:
+        return []
+    # TODO: support counting of other fields (# of unique users, ...)
+    kwargs = {
+        "dataset": dataset,
+        "start": start,
+        "end": end,
+        "filter_keys": {"project_id": [project.id]},
```

Comment:
Might wanna update `filter_keys` with the same conditional check for transactions.

Reply:
are you talking about something like this below?
```
if dataset == Dataset.Transactions:
        kwargs["conditions"] = [[["has", ["group_ids", group_id]], "=", 1]]
```